### PR TITLE
Remove estimator modules and its dependencies

### LINKED ISSUE #143: 1.0 Release - IO and Statistics, remove experiments and model training
The current project supports features like ML experimentation, dataset IO, stats, and analysis on the notebook. In this release, we will remove all dependencies and code related model training and experiments to keep datasetinsights focused on io and statistics. This will reduce heavy decencies such as pytorch or tensorflow to make it easier for user to consume data from perception SDK without taking heavy dependencies. 

On high level, the 1.0 release of datasetinsights will contain the following modules

```
datasetinsights/
   io/
   stats/
   datasets/unity_perception/
notebooks/
```

## Checklist

- [x] Remove estimators, configs, trackers, kubflow pipeline, checkpoint
- [x] Prune dependencies 
- [x] Update README to reflect the current high level API. 
- [x] Update Perception_Statistics notebook, Remove yacs
- [x] Fix github action issue to prevent push 0.2.x tag to latest tag on dockerhub.

## How do I access model training code during this refactoring?

- Create a new branch after “Coco cat training and evaluation” commit, name 0.2.x.
- Create a tag on the master branch with the name 0.2.4. 
- Experimentation module: Users who want to use the train, evaluate can build the docker image on the code tagged with 0.2.4.
- Dataset IO, stats, and analysis on the notebook: Use the latest code from the master branch.

Any new changes or enhancements for the experimentation module will be done in the new repo but if any critical bug comes, checkout code from 0.2.x, fix the bug, and merge back to 0.2.x branch.
Release new tag from 0.2.x branch with the name 0.2.4.x

## What's the plan for future model training code? 

We will create a new repo that contain model training /experiment code. 
[Link to the new repo once it's ready]

## This work is split into multiple pull requests

- PR https://github.com/Unity-Technologies/datasetinsights/pull/142: Includes removal/fixes of the following modules which are related to model training and evaluation. 
Estimators
Train and Evaluate Command
Evaluation notebook
Configs yaml
Fixed docs rst and toctree dependency
IO module:
Tracker
Kfp output
Checkpoint
Config handler
Summary writer
Kubeflow pipelines.py
Evaluation metric
Corresponding unit test
docs

- PR https://github.com/Unity-Technologies/datasetinsights/pull/147: 
Remove tf and torch dependencies
Remove base and any public dataset
Remove any redundant library
Remove loader, draft notebook, protos
Updated readme
Updated docs

- PR https://github.com/Unity-Technologies/datasetinsights/pull/145: Documentation fixes.

- PR https://github.com/Unity-Technologies/datasetinsights/pull/154: Notebook fixes and update read me with instructions to show how to use this package, remove yacs
- PR https://github.com/Unity-Technologies/datasetinsights/pull/155: Make sure release from 0.2.x branch should not override docker latest tag.

### PR DESCRIPTION
**Pull request by 86sanj**
# Peer Review Information

This commit includes deletion/fixes of following modules/files:
- Estimators
- Train and Evaluate Command
- Evaluation notebook
- Configs yaml
- IO module:( Tracker, Kfp output, Checkpoint, Config handler, Summary writer)
- Kubeflow pipelines
- Evaluation metric
- Corresponding unit test cases

Requirement: https://github.com/Unity-Technologies/datasetinsights/issues/143

# Pull Request Check List

- [x] Added **tests** for changed code.
- [x] Updated **documentation** for changed code.

### CODE REVIEW

**Comment by Saurav-D** [comment]
Can add more information about the purpose of this PR ?

**Comment by adason** [comment]
The requirements doc in the description is private doc. Can you copy the content of this doc to github? Maybe create an issue describing the plan to remove experiment run? 

Can you also update the title of this PR to make sure it's clear what this PR is trying to do?

File: pyproject.toml [hunk 1]
```diff
@@ -48,7 +48,6 @@ dask = {extras = ["complete"], version = "^2.14.0"}
 dash = "1.12.0"
 tensorflow = "^2.2.0"
```
Comment (adason) [inline pyproject.toml:49]: Should we remove tensorflow dependency? Can you check if some of the dependencies can be removed? Can you generate a new poetry.lock file with the updated dependencies?

File: docs/source/datasetinsights.rst [hunk 2]
```diff
@@ -7,7 +7,6 @@ datasetinsights
 
    datasetinsights.commands
    datasetinsights.datasets
-   datasetinsights.estimators
    datasetinsights.evaluation_metrics
```
Comment (adason) [inline docs/source/datasetinsights.rst]: This should be removed as `datasetinsights.evaluation_metrics` are removed from the package.

File: pyproject.toml (hunk 1 above)
Comment (86sanj) [reply pyproject.toml:49]: Will do this in next PR.

**Review by adason — APPROVED** [review]

**Review by Saurav-D — APPROVED** [review]
